---
layout: default
title: splicing
---

# I presented a poster at ISMB2018 on negative selection and unbiased benchmarking of splicing predictors

ABSTRACT: Presence or absence of genetic variation in the general population is reflective of negative selection (particularly for genes predicted to be haploinsufficient) and mutation probability. In this work, we used this property to evaluate changes in splicing regulatory sequence. We considered all genomic positions covered by gnomAD 120,000 exomes. We then looked at variance in negative selection with respect to mutation type corrected for nucleotide mutation probability. We observed 10% of all possible synonymous variants and 5.5% of all possible missense as well as 1% of all nonsense mutations (stop gains, splice-acceptor,splice-donor). Based on selection constraint methodology we present an unbiased approach for evaluating mis-splicing predictors. MaxEntScan and DeepScan,a deep learning tool, are the best splicing variant effect predictors, when considering only variants in the splicing consensus region. However, splicing effects in the distal intronic and exonic regions appear to be too weak to be detected. In conclusion we present an investigation into impact of negative selection with regards to mutation type as well as an alternative approach for splicing predictor evaluation that trades a reduction in statistical power for an unbiased evaluation set. [link](https://www.iscb.org/cms_addon/conferences/ismb2018/posters.php?track=RNA&session=A)

<object data="../pdf/ismb_poster.pdf" type="application/pdf" width="700px" height="700px">
    <embed src="../pdf/ismb_poster.pdf">
        <p> <ul> Link to poster: This browser does not support PDFs. Please download the PDF to view it: <a href="../pdf/ismb_poster.pdf">Download PDF</a>. </ul> </p>
    </embed>
</object>
